### PR TITLE
Dask masked fix

### PR DESCRIPTION
**Pull request by pp-mo**
Bring iris_grib up-to-date with recent changes to Iris, 
to use masked-arrays instead of nan masking in Dask (Dask v 0.13).

### CODE REVIEW

File: iris_grib/__init__.py [hunk 1]
```diff
@@ -145,7 +145,6 @@ class GribWrapper(object):
     def __init__(self, grib_message, grib_fh=None):
         """Store the grib message and compute our extra keys."""
         self.grib_message = grib_message
-        self.realised_dtype = np.array([0.]).dtype
```
Comment (pelson) [inline iris_grib/__init__.py]: API change

File: iris_grib/__init__.py [hunk 2]
```diff
@@ -698,6 +694,10 @@ def _message_values(grib_message, shape):
     data = gribapi.grib_get_double_array(grib_message, 'values')
     data = data.reshape(shape)
 
+    # Handle missing values in a sensible way.
```
Comment (pelson) [inline iris_grib/__init__.py]: Is this a hangover from not having masks?

File: iris_grib/_save_rules.py [hunk 3]
```diff
@@ -1157,17 +1157,20 @@ def product_definition_section(cube, grib):
...
+            # Use the data's fill value.
+            fill_value = float(cube.data.fill_value)
+        else:
             # We can't use the cube's fill value if it's NaN,
             # the GRIB API doesn't like it.
             # Calculate an MDI outside the data range.
             min, max = cube.data.min(), cube.data.max()
             fill_value = min - (max - min) * 0.1
+        # Prepare the unmasked data array, using fill_value as the MDI.
+        data = cube.data.filled(fill_value)
```
Comment (pelson) [inline iris_grib/_save_rules.py]: Again, is this a hangover?

File: iris_grib/__init__.py (hunk 1 above)
Comment (pp-mo) [reply iris_grib/__init__.py]: Well, not if we never released the previous version, no ? ;-)

File: iris_grib/__init__.py (hunk 2 above)
Comment (pp-mo) [reply iris_grib/__init__.py]: Not totally sure.
This is simply reflecting what is currently in Iris master.

File: iris_grib/_save_rules.py (hunk 3 above)
Comment (pp-mo) [reply iris_grib/_save_rules.py]: As above...

File: iris_grib/__init__.py (hunk 2 above)
Comment (pelson) [reply iris_grib/__init__.py]: I'd love to find out more...

**Pull request merged**